### PR TITLE
Implement data structure

### PR DESCRIPTION
**Pull request by AltNico**
Based on @xamanu's commit I finished the work on osm2gtfs' data structure #30. I'm currently testing with fenix' creator (got banned one time already by Overpass 🙂) and it seems to work. I hope you have good tests testing the creators too. I haven't tested incofer yet but it should work as it has no dependencies to the newly introduced classes.

### CODE REVIEW

**Comment by grote** [comment]
Wow awesome! This came unexpected! Thanks a lot for finishing that work!

Sine this touches most of the data structures it looks like it will conflict with the other two MRs. Do you mind us merging these first and then getting to yours?

**Comment by AltNico** [comment]
@grote 
> Sine this touches most of the data structures it looks like it will conflict with the other two MRs. Do you mind us merging these first and then getting to yours?

Sure, feel free to merge all the other stop first. I will rebase and fix stuff later 👍

**Comment by grote** [comment]
@AltNico I think this is now good to be rebased.

File: creators/trips_creator.py [hunk 1]
```diff
@@ -13,6 +13,18 @@ def __repr__(self):
...
     def add_trips_to_schedule(self, schedule, data):
+        """
+        route_id  # Required: From Line
+        service_id  # Required: To be generated
+        trip_id  # Required: To be generated
+
+        trip_headsign # Itinerary "to"
+        direction_id  # Order of tinieraries in Line object
+        wheelchair_accessible  # Itinerary "wheelchair_accessible"
+        bikes_allowed: # Itinerary "bikes_allowed"
```
Comment (grote) [inline creators/trips_creator.py]: extra colon?

File: creators/routes_creator.py [hunk 2]
```diff
@@ -13,4 +13,16 @@ def __repr__(self):
...
+        """
+        route_id  # Required: From Line
+        route_type  # Required: From Line
+
+        route_short_name  # Required: To be generated from Line or Itinerary
+        route_long_name  # Required: To be generated from Line or Itinerary
+
+        route_desc # From Line
+        route_url  # From Line
+        route_color: # From Line
```
Comment (grote) [inline creators/routes_creator.py]: extra colon?

File: core/osm_connector.py [hunk 3]
```diff
@@ -259,48 +270,63 @@ def get_stops(self, refresh=False):
...
-    def _build_route_variant(self, route_variant, query_result_set, rm=None):
-        """Helper function to build a RouteVariant object
+        if 'route_master' in route_master.tags:
+            route_type = route_master.tags['route_master'].capitalize()
+
+        # If there was no route_master present we have a route relation here
+        elif 'route' in route_master.tags:
+            route_type = route_master.tags['route'].capitalize()
+
+        print route_type
```
Comment (grote) [inline core/osm_connector.py]: stray print? Btw. please let's aim for python 3 already so the transition will be easy.

File: creators/accra/trips_creator_accra.py [hunk 4]
```diff
@@ -18,7 +19,7 @@ def add_trips_to_schedule(self, schedule, data):
 
         lines = data.routes
         for route_ref, line in sorted(lines.iteritems()):
-            if type(line).__name__ != "RouteMaster":
+            if not isinstance(line, Line):
```
Comment (grote) [inline creators/accra/trips_creator_accra.py]: Much better! 👍

Comment (AltNico) [reply creators/accra/trips_creator_accra.py]: Thanks for the first quick feedback. I pushed a new commit with fixes but sadly it does not show a cool diff between the different versions of the pull request like in GitLab 🙁

**Comment by nlehuby** [comment]
Hello,

I won't have time to do a full review before a few days, but there are some issues with the Accra creator (missing class attributes at least).

**Comment by AltNico** [comment]
@nlehuby 
> I won't have time to do a full review before a few days, but there are some issues with the Accra creator (missing class attributes at least).

Ah, you're right, I forgot to change that. I amended the commit to correct that.

**Comment by xamanu** [comment]
I just reviewed this. Looks very good so far! Thanks for finishing this up, @AltNico. I some minor issues:

* Needs to be rebased again (there is a simple conflict in the `setup.py` now)
* A little fix for the accra generator ([see here](https://github.com/xamanu/osm2gtfs/commit/9ac60db339e0bf42bce618ce7f04309f73a7526b#diff-bcd3a8e491ea3f3bee3ea782354fa76eL64))
* I suggest to not include the unused `debug.py`, as this is probably better handled with the issue #46 (and [ongoing work here](https://github.com/xamanu/osm2gtfs/tree/issue-46-debug))

All I found is suggested as [a pull request](https://github.com/AltNico/osm2gtfs/pull/1) to @AltNico, wich hopefully makes it easy to add to this pull request here :smile_cat:

**Comment by nlehuby** [comment]
@xamanu your patch on the Accra creator is not ok : it alters the structure of the GTFS stop_id.

**Comment by xamanu** [comment]
@AltNico asked me to re-request this PR newly (https://github.com/AltNico/osm2gtfs/pull/1). So there is a new pull request substituting this one: https://github.com/grote/osm2gtfs/pull/84

**Pull request closed without merging**